tests/e2e/scale/test_scale_small_file_workload.py

```diff
@@ -0,0 +1,120 @@
+"""
+Test to exercise Small File Workload
+
+"""
+
+# Builtin modules
+import logging
+import time
+
+# 3ed party modules
+import pytest
+
+# Local modules
+import ocs_ci.ocs.resources.pod as pod
+from ocs_ci.ocs.ripsaw import RipSaw
+from ocs_ci.ocs import constants, node
+from ocs_ci.framework.testlib import E2ETest, scale
+from ocs_ci.ocs.small_file_workload import smallfile_workload
+from ocs_ci.ocs.elasticsearch import ElasticSearch
+
+log = logging.getLogger(__name__)
+
+
+def get_cephfs_data():
+    """
+    Look through ceph pods and find space usage on all cephfilesystem pods
+
+    Returns:
+        Dictionary of byte usage, indexed by pod name.
+    """
+    ceph_pod = pod.get_ceph_tools_pod()
+    ceph_status = ceph_pod.exec_ceph_cmd(ceph_cmd="ceph df")
+    ret_value = {}
+    for pool in ceph_status["pools"]:
+        if "cephfilesystem" in pool["name"]:
+            ret_value[pool["name"]] = pool["stats"]["bytes_used"]
+    return ret_value
+
+
+@pytest.fixture(scope="module")
+def scale_leaks(request):
+    """
+    For scale tests, get bytes used by cephfs at the start,
+    and compare with the cephfs usage at the finish.
+    """
+
+    def teardown():
+        """
+        After all the ripsaw creation and deltion runs have finished,
+        test to see if there was a leak in storage.
+        """
+        log.info("In scale_leaks teardown")
+        time.sleep(120)
+        orig_data = pytest.ceph_fs_use
+        now_data = get_cephfs_data()
+        for entry in now_data:
+            log.info(f"{entry} now uses {now_data[entry]} bytes")
+            log.info(f"{entry} originally used {now_data[entry]} bytes")
+            check = (now_data[entry] - orig_data[entry] < 1000000000,)
+            #
+            # Maybe we should do some more detailed checking here.
+            # For now, report an error if there is 1G that appears to leak.
+            #
+            errmsg = f"{entry} over 1G larger -- possible leak"
+            assert check, errmsg
+
+    pytest.ceph_fs_use = get_cephfs_data()
+    request.addfinalizer(teardown)
+
+    return scale_leaks
+
+
+@pytest.fixture(scope="function")
+def es(request):
+
+    # Create internal ES only if Cloud platform is tested
+    if node.get_provider().lower() in constants.CLOUD_PLATFORMS:
+        es = ElasticSearch()
+    else:
+        es = None
+
+    def teardown():
+        if es is not None:
+            es.cleanup()
+            time.sleep(10)
+
+    request.addfinalizer(teardown)
+    return es
```

Comment:
you can use the general fixure from  tests.conftest.py (line 3531)

Reply:
Done.  I have pushed a new version.

Reply:
The performance tests are still run in tests/e2e/performance/test_small_file_workload.py   After this PR is done, and after the  Elastic search exception that I reported on issue 3705 gets resolved, I will adapt the performance small file test to also use the ocs_ci/ocs/small_file_workload.py functions.